packages/jumpstarter/jumpstarter/client/grpc.py

```diff
@@ -250,3 +254,25 @@ async def DeleteLease(self, *, name: str):
                     name="namespaces/{}/leases/{}".format(self.namespace, name),
                 )
             )
+
+
+@dataclass(frozen=True, slots=True)
+class MultipathExporterStub:
+    channels: InitVar[list[Channel]]
+
+    __stubs: dict[Channel, Any] = field(init=False, default_factory=OrderedDict)
+
+    def __post_init__(self, channels):
+        for channel in channels:
+            stub = SimpleNamespace()
+            jumpstarter_pb2_grpc.ExporterServiceStub.__init__(stub, channel)
+            router_pb2_grpc.RouterServiceStub.__init__(stub, channel)
+            self.__stubs[channel] = stub
+
+    def __getattr__(self, name):
+        for channel, stub in self.__stubs.items():
+            # find the first channel that's ready
+            if channel.get_state(try_to_connect=True) == ChannelConnectivity.READY:
```

Comment:
```suggestion
            if channel.get_state(try_to_connect=True) == ChannelConnectivity.READY:
```

Do we want to add some debug logging here when we try?